adapter/serviceregistry/aggregate/controller.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregate
+
+import (
+	"github.com/golang/glog"
+
+	"istio.io/pilot/model"
+	"istio.io/pilot/platform"
+)
+
+// Controller aggregates data across different registries and monitors for changes
+type Controller struct {
+	registries map[platform.ServiceRegistry]model.Controller
```

Comment:
The logic below expects the `registries` map values to implement `model.Controller`, `model.ServiceDiscovery`, and `model.ServiceAccounts` or there will be a panic. If that is the requirement, I instead recommend creating an interface that requires all three interfaces to be implemented:

```golang
type Registry interface {
  model.Controller
  model.ServiceDiscovery
  model.ServiceAccounts
}
```
and then
```golang
type Controller struct {
  registries map[platform.ServiceRegistry]Adapter
}
```